src/main/scala/at/logic/skeptik/expression/formula/formulas.scala

```diff
@@ -57,9 +64,41 @@ object Atom extends Formula {
     require(atom.t == o)
     atom
   }
+  def apply(name: String, args: List[E]) = {
+    def createType(list : List[E]) : T = list match {
+      case Nil   => o
+      case e::es => e.t -> createType(es)
+    }
+    val p    = Var(name,createType(args))
+    val atom = AppRec(p,args)
+    require(atom.t == o)
+    atom
+  }
   def unapply(e:E) = e match {
     case AppRec(f,args) if (e.t == o && !isLogicalConnective(f)) => Some((f,args))
     case _ => None
   }
 }
 
+object ConditionalFormula extends Formula{
```

Comment:
Sorry, I just now read this comment.
I will do it in that way. I will change this in my previous definitions as I see them from now on too.